src/store.c

```diff
@@ -351,14 +352,18 @@ int pthreads_store_write(zval *object, zval *key, zval *write) {
 				normal refcounting, we'll just never read the reference
 			*/
 			rebuild_object_properties(&threaded->std);
+			
+			pthreads_store_sync(object);
```

Comment:
Testing the memory leak isn't really necessary, since it's quite unlikely a regression will be introduced here. I just wasn't sure how syncing the pthreads store linked to the memory leaks (it doesn't, but I see that you're using this PR to target different bugs).